x-pack/plugins/apm/public/components/shared/kuery_bar/index.tsx

```diff
@@ -151,18 +151,18 @@ export function KueryBar(props: {
         return;
       }
 
-      if (typeof props.onSubmit === 'function') {
```

Comment:
what are the consequences of moving this function to after pushing to `history`? I assume this was intentional. E.g., the SelectServices component uses this, and I don't think we want the kuery to be persisted to the URL there.

Reply:
Good point. It does break `SelectServices`. I'll add a new boolean prop to `KueryBar` called `urlPersist` to conditionally push to history.

Reply:
I don't get why it's necessary though? If you don't pass onSubmit, which you are not doing for the service map, it should persist the query parameter to the URL. Am I missing something? Also it's not just SelectServices, there's another component too.

Reply:
I see what you mean, the prop `onSubmit` should be sufficient with the check and early return before the history push. I'll change it back to how it was before.